makefiles/cflags.inc.mk

```diff
@@ -67,7 +67,8 @@ endif
 # Check if linker supports `-z noexecstack`. Handle BUILD_IN_DOCKER separately,
 # as this is run in the host environment rather than inside the container. We
 # just hardcode this in the BUILD_IN_DOCKER case for now.
-LINKER_SUPPORTS_NOEXECSTACK ?= $(shell echo "int main(){} void _exit(int n) {(void)n;while(1);}" | LC_ALL=C $(LINK) -xc - -o /dev/null -lc -Wall -Wextra -pedantic -z noexecstack 2> /dev/null && echo 1 || echo 0)
+LINKER_SUPPORTS_NOEXECSTACK ?= $(shell LC_ALL=C $(LINK) $(RIOTTOOLS)/testprogs/minimal_linkable.c -o /dev/null -lc -Wall -Wextra -pedantic -z noexecstack 2> /dev/null && echo 1 || echo 0)
+$(info LINKER_SUPPORTS_NOEXECSTACK="$(LINKER_SUPPORTS_NOEXECSTACK)")
```

Comment:
```suggestion
```

Good thing that the KConfig test also fails when there still is unexpected debug output that was forgotten to be removed 😅